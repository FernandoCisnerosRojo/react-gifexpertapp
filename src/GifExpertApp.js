import React from 'react'
import { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <div>
            <h2>GifExpertApp</h2>
            <hr></hr>

            <AddCategory setCategories={setCategories} />

            <ol>
                {
                    categories.map(category => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </div>
    )
}

export default GifExpertApp
