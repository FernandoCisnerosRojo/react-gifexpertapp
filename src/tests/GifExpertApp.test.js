import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('GifExpertApp Tests', () => {
    test('should match snapshot', () => {
        const wrapper = shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot();
    })

    test('should match snapshot with categories', () => {
        const categories = ['car', 'bike'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })

})
