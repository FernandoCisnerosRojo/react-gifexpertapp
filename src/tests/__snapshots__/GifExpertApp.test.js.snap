// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Tests should match snapshot 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</div>
`;

exports[`GifExpertApp Tests should match snapshot with categories 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="car"
      key="car"
    />
    <GifGrid
      category="bike"
      key="bike"
    />
  </ol>
</div>
`;
